### DISCUSSION TITLE
Trame+Paraview memory footprint(usage) grew up

### DISCUSSION THREAD

**Discussion by khumuunee** [Kitware/trame#135] [Q&A] [answered]
Hello,

I'm trying to visualize 3d Unstructured grid with large datasets on trame+paraview. I have problem with memory footprint grow up when rotate object (zoom object, move object, ...).  When i working on paraview(with vtk file) no problem with memory increase. Is there something to memory release or clear? How can i solve this memory increase problem?

Thanks!

**Comment by khumuunee** [comment]
Memory usage increase is same on Trame/VTK example app_cone.py too.

**Reply by jourdain** [reply]
Thanks for reporting that!

**Comment by jourdain** [comment]
Could you be more specific in term of memory increase (amount, python process, web browser, ...) and which code exactly are you running. 

When trying `./examples/06_vtk/01_SimpleCone/RemoteRendering.py` I don't see anything wrong on my side using macOS and Chrome.

**Reply by khumuunee** [reply]
Thanks for the reply!

This is my working environment:

OS                    ->  Windows 8.1 Pro 64-bit
RAM                 -> 16GB
CPU                  -> i5-6400
Python              -> 3.9.13
Trame               -> 2.2.0
VTK                   -> 9.2.2
Chrome            -> 106.0.5249.120

There was a problem with the RemoteRendering method. 
My goal is to visualize a large dataset 3d unstructured grid (Millions of cells. 20+ variables per cell). So trame/paraview is more suitable for this. Because when I tried with trame/vtk (both of Local or Remote Rendering) on large dataset, it crashed. When working with trame/paraview it rendering without crash, but I have the same memory increase problem as trame/vtk remote rendering.

I recorded the аction:

https://user-images.githubusercontent.com/76699571/198181696-9444c6cb-9ae5-49bb-9460-e21a24a3fefc.mp4

**Comment by jourdain** [comment, marked as answer]
Wow, indeed that is crazy. I don't see that on my mac. Which version of wslink do you have?

**Reply by jourdain** [reply]
Actually I see that issue with VTK 9.2 but not with VTK 9.1

**Reply by khumuunee** [reply]
That's right. I tried Trame/VTK with VTK 9.1, it is ok. But with Trame/Paraview, it is the same as before. It seems to be due to the version of Paraview. Version is 5.11. As I understand it, VTK is inside Paraview. Which version of Paraview should i have to make it no issue?

Thank you for taking the time to do this!

**Reply by jourdain** [reply]
PV 5.10 should be ok.

**Reply by khumuunee** [reply]
Working with Paraview 5.10 i have another problem.
When run cmd command:

pvpython D:\Work\Crate\Project\Trame\trame-tutorial\05_paraview\SimpleCone.py --venv D:\Work\Crate\Project\Trame\trame-tutorial\.pvenv

ModuleNotFoundError: No module named 'trame'

![error](https://user-images.githubusercontent.com/76699571/198358203-46a44218-b652-4545-8dd4-1260fd4c25e7.png)

**Reply by jourdain** [reply]
That is expected as `trame` is supposed to come with a local `venv` that you setup. 
See the readme [here](https://github.com/Kitware/trame/tree/master/examples/07_paraview)

**Reply by khumuunee** [reply]
This helped me to solve ModuleNotFoundError:
https://github.com/Kitware/trame/discussions/99#discussioncomment-3262224

And with Paraview 5.10 have no memory increase.

Thank you again!

**Reply by jourdain** [reply]
Glad we found a solution as well as detecting a memory problem (even if it is a bit sad...)
